src/rules/function-url-no-scheme-relative/index.js

```diff
@@ -0,0 +1,55 @@
+import valueParser from "postcss-value-parser"
+import {
+  isStandardSyntaxValue,
+  isVariable,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "function-url-no-scheme-relative"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: "Unexpected scheme relative",
+})
+
+export default function (actual) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    root.walkDecls(function (decl) {
+      valueParser(decl.value).walk(valueNode => {
+        if (valueNode.type !== "function"
+          || valueNode.value.toLowerCase() !== "url"
+          || !valueNode.nodes.length > 0
+        ) { return }
+
+        const urlValueNode = valueNode.nodes[0]
+
+        if (!urlValueNode.value
+          || !isStandardSyntaxValue(urlValueNode.value)
+          || isVariable(urlValueNode.value)
+        ) { return }
+
+        const urlValue = urlValueNode.value
+
+        // Ignore all url which less two characters, because scheme relative always start with `//`
+        if (urlValue.toLowerCase().indexOf("data:") === 0
+          || urlValue.length < 2
```

Comment:
This condition is also unnecessary, I think, because you are checking the same below. If the motivation is speed optimization, I think the next too checks are too trivial to require any such optimization.